### DISCUSSION TITLE
Adjust cloud storage replica polling interval

### DISCUSSION THREAD

**Discussion by netthier** [cloudnative-pg/cloudnative-pg#2807] [Q&A] [answered]
I have configured a cluster with an `externalCluster` in Azure Blob Storage and `replica.enabled: true`.
However, I quickly noticed that the cluster executed `ListBlobs` every few seconds, despite the remote only uploading WALs every 10 minutes.
As `ListBlobs` requests aren't cheap and they're being made way too often, is there a way to increase the interval in which the remote storage is being polled?

**Comment by leonardoce** [comment, marked as answer]
Hi!

TLDR: you can use the [`wal_retrieve_retry_interval`](https://www.postgresql.org/docs/16/runtime-config-replication.html#GUC-WAL-RETRIEVE-RETRY-INTERVAL) PostgreSQL parameter to configure that polling timeout.

The `ListBlobs` request you refer to is triggered by the `restore_command` PostgreSQL setting.
CNPG uses [barman-cloud-wal-restore](https://docs.pgbarman.org/release/2.11/barman-cloud-wal-restore.1.html) for that, and `barman-cloud-wal-restore` does that.
PostgreSQL is calling the `restore_command` to check if there are new WAL files, and despite you uploading WALs every 10 minutes, PostgreSQL 15 will do that every 5 seconds by default.
You can see that [in the source code](https://github.com/postgres/postgres/blob/e5975c2daad0327b9faa6b62bcfbf173680d14b9/src/backend/access/transam/xlogrecovery.c#L3611).

The `wal_retrieve_retry_interval` setting allows you to customize that, at the expense of a less "reactive" log shipping replica.

**Reply by netthier** [reply]
Thanks for the answer, tweaking that setting did indeed fix this issue.
However, every time a WAL is retried, I observe 50-70 `ListBlobs` requests for some reason, which is way more than expected. Do you happen to know why so many of these requests occur?
For now I've just set that config to an even higher value, which works just fine as well.
